File: my_py_pkg/my_py_pkg/hw_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from my_robot_interfaces.msg import HardwareStatus

class hardwarestatuspublishernode(Node):
    def __init__(self):
        super().__init__('hardware_status_publisher')
        self.hw_status_publisher = self.create_publisher(HardwareStatus, 'hardware_status', 10)
        self.timer=self.create_timer(1, self.publish_hw_status)
        self.get_logger().info('Hardware Status Publisher has been started')
    
    def publish_hw_status(self):
        msg=HardwareStatus()
        msg.temperature=45
        msg.are_motors_ready=True  
        msg.debug_message='Everything is OK'
        self.hw_status_publisher.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = hardwarestatuspublishernode()
    rclpy.spin(node)
    rclpy.shutdown()
    
if __name__ == '__main__':
    main()

File: my_cpp_pkg/src/hw_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

class hwstatuspublisher : public rclcpp::Node
{
    public:
        hwstatuspublisher() : Node("hw_status_publisher")
        {
            publisher_= this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hw_status", 10);
            timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&hwstatuspublisher::publish_hw_status, this));
            RCLCPP_INFO(this->get_logger(), "HWStatusPublisher has been started.");
        }
    private:
        void publish_hw_status()
        {
            auto msg = my_robot_interfaces::msg::HardwareStatus();
            msg.temperature = 45;
            msg.are_motors_ready = true;
            msg.debug_message = "Everything is running fine.";
            publisher_->publish(msg);
        }
        rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<hwstatuspublisher>());
    rclcpp::shutdown();
    return 0;
}